tests/api/core/content-manager/api/document-metadata.test.api.js

```diff
@@ -174,4 +234,140 @@ describe('CM API - Document metadata', () => {
+      },
+      price: {
+        expectation: expect.any(Number),
+        getDefaultValue: () => 1,
+      },
+      features: {
+        expectation: expect.any(Object),
+        getDefaultValue: (locale) => ({
+          name: `Feature 1 ${locale}`,
+          description: 'Description 1',
+          slogan: 'Slogan 1',
+        }),
+      },
+      createdAt: {
+        expectation: expect.any(String),
+      },
+      updatedAt: {
+        expectation: expect.any(String),
+      },
+      name: {
+        expectation: expect.any(String),
+      },
+    };
+
+    const unWantedKeys = {
+      // These fields have no validation requirements
+      // We will check that they are not included in the available locales response
+      shopName: {
+        expectation: expect.any(String),
+        getDefaultValue: (locale) => `BuyMyStuff-${locale}`,
+      },
+    };
+
+    const getExtraContent = (locale) => {
+      return {
+        ...Object.entries({ ...wantedKeys, ...unWantedKeys }).reduce(
+          (acc, [key, { getDefaultValue }]) => {
+            if (getDefaultValue === undefined) {
+              // If there is no default value function, do not include the key
+              return acc;
+            }
+
+            return { ...acc, [key]: getDefaultValue(locale) };
+          },
+          {}
+        ),
+      };
+    };
+
+    // Create products with different locales with content for every kind of
+    // field that requires validation
+    const defaultLocaleProduct = await rq({
+      method: 'POST',
+      url: `/content-manager/collection-types/${PRODUCT_UID}`,
+      body: {
+        name: `prod-en-draft`,
+        ...getExtraContent('en'),
+      },
+    });
+    const documentId = defaultLocaleProduct.body.data.documentId;
+
+    await rq({
+      method: 'PUT',
+      url: `/content-manager/collection-types/${PRODUCT_UID}/${documentId}`,
+      body: {
+        name: `prod-${extraLocaleCode}-draft`,
+        ...getExtraContent(extraLocaleCode),
+      },
+      qs: { locale: extraLocaleCode },
+    });
+
+    const draftProduct = await getProduct(documentId, 'en', 'draft');
+    const { meta } = await formatDocument(PRODUCT_UID, draftProduct, {});
+
+    expect(meta.availableLocales).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining(
+          Object.keys(wantedKeys).reduce(
+            (acc, key) => ({ ...acc, [key]: wantedKeys[key].expectation }),
+            {}
+          )
+        ),
+      ])
+    );
+
+    expect(meta.availableLocales).toEqual(
+      expect.arrayContaining([
+        expect.not.objectContaining(
+          Object.keys(unWantedKeys).reduce(
+            (acc, key) => ({ ...acc, [key]: unWantedKeys[key].expectation }),
+            {}
+          )
+        ),
+      ])
+    );
+
+    expect(meta.availableStatus).toEqual([]);
+  });
+
+  test('Does not return any sensitive information in the available status', async () => {
```

Comment:
I wonder if either this or the previous test could use `toEqual` / `toContainEqual` to ensure we don't get any unexpected data? I know that kind of test is more of a pain to write and keep updated as things change, but in this case it might be better